eng/pipelines/spec-gen-sdk-net.yml

```diff
@@ -0,0 +1,51 @@
+parameters:
+  - name: SpecRepoBranch
+    type: string
+    default: 'main'
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoBranch
+    type: string
+    default: 'main'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'OpenAPI'
+    default: 'TypeSpec'
+    displayName: 'Configuration Type'
+  - name: TspConfigPath
+    type: string
+    default: 'specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml'
+    displayName: 'Path to TypeSpec Configuration file'
+  - name: ReadmePath
+    type: string
+    default: 'specification/contosowidgetmanager/data-plane/readme.md'
+    displayName: 'Path to OpenAPI Configuration'
+  - name: ApiVersion
+    type: string
+    default: 'not-specified'
+    displayName: 'API Version of Spec to be used for SDK Generation'
+  - name: SkipPullRequestCreation
+    type: boolean
+    default: false
+
+trigger: none
+pr: none
+
+variables:
+  SdkRepoUrl: 'https://github.com/Azure/azure-sdk-for-net'
```

Comment:
Is the only difference this variable? Seems like a lot of duplication.

Reply:
I was chatting with Ray and Dozie and we were trying to have any language specific values hard -coded in these entry point files. However, if we aren't going to use them and we are going to depend on `$(Build.Repository.Name)` then perhaps we should revisit and only have one yml file for all the languages. It doesn't even appear that we are using `SdkRepoUrl` currently.

Sorry @raych1 and @chidozieononiwu if my guidance was pushing the other way, but I thought we would have other hard-coded variables per-language.

Reply:
A single yaml seems sufficient, then the SdkRepoUrl can be set on the pipeline UI

Reply:
`SdkRepoUrl` is used in the `Create Run Time Variables` step of the referenced pipeline template. We still need multiple yml files as `Build.Repository.Name` only controls the spec repo.

Reply:
Do we expect SDKRepoUrl to be the only unique variable we hard-code for a specific language? If so, perhaps we can simply make it a pipeline UI variable. 

In general, I like having those things be in the yml files so we can easily discover them and have them under source control, but if it is only the one variable perhaps, we can have it as a UI variable and only have the one entry point yml file.

Reply:
At the time `SdkRepoUrl` is the only unique variable. Perhaps we can add one language as the default then add a comment indicating that it is overwritten in the pipeline UI for each language.

Reply:
Just to clarify again, should it be single or multiple yaml configurations

Reply:
I would make it a single configuration. And you can default the template parameter to `$(SDKRepoUrl)`. I just wanted to make sure it is clear that this is a parameter to this template and would allow use to split this if needed.